saleor/order/migrations/tasks/saleor3_18.py

```diff
@@ -0,0 +1,55 @@
+from django.db.models import F, Func, OuterRef, Subquery
+
+from ....celeryconf import app
+from ...models import Order, OrderLine
+
+# celery task for one batch took 0.1s and 10.54MB at its peak
+BATCH_SIZE = 1000
+
+
+@app.task
+def update_order_subtotals(start_number=None):
```

Comment:
I don't think that this will work in its current form. 

---

```
In [67]: Order.objects.count()
Out[67]: 22
Number of queries 1

In [68]: Order.objects.first().number
Out[68]: 22
Number of queries 1

In [69]: Order.objects.last().number
Out[69]: 1
Number of queries 1
```

I have 22 orders on my local machine.

---

```
In [56]: start_number=None
In [57]: if start_number is None:
    ...:     first_order = Order.objects.last()
    ...:     if first_order is None:
    ...:         print("None orders")
    ...:     start_number = first_order.number
    ...:
Number of queries 1

In [58]: start_number
Out[58]: 1
```
First call gives me start_number as 1

---

The current batch gives me:

```
In [60]: current_batch_order_numbers = Order.objects.filter(
    ...:     number__gt=start_number
    ...: ).values_list("number", flat=True)[:BATCH_SIZE]

In [61]: current_batch_order_numbers
Out[61]: <OrderQueryset [22, 21, 20, 19, 18]>
```

---

Then at the end you have:
```
last_number = current_batch_order_numbers[0]
```

Which gives me:
```
In [62]: last_number = current_batch_order_numbers[0]
Number of queries 1

In [63]: last_number
Out[63]: 22
```

So next iteration of the task, will trigger the task with number 22. This will result with:

```
In [64]: start_number = last_number

In [65]: current_batch_order_numbers = Order.objects.filter(
    ...:     number__gt=start_number
    ...: ).values_list("number", flat=True)[:BATCH_SIZE]

In [66]: current_batch_order_numbers
Out[66]: <OrderQueryset []>
```